lib/launch.c

```diff
@@ -150,24 +150,36 @@ struct stage3_entry {
 	unsigned int     lcore_id;
 };
 
+struct stage3_lcores_info {
+	char     *names[RTE_MAX_LCORE];
+	uint16_t num_lcores;
+};
+
+static struct stage3_lcores_info lcores_info;
```

Comment:
There's no need for `lcores_info`, `get_filtered_num_lcores()` can simply go over list `launch_heads.stage3`.